python_repository/repository.py

```diff
@@ -132,10 +131,10 @@ def _entity_class(cls) -> Type[GenericEntity]:
         if not issubclass(entity_class, SQLModelEntity):
             raise TypeError(f"Entity class {entity_class} for {cls.__name__} must be a subclass of {SQLModelEntity}")
 
-        return entity_class  # TODO: Expression of type "Type[SQLModelEntity]" cannot be assigned to return type "Type[GenericEntity@BaseRepository]"
+        return entity_class  # type: ignore
```

Comment:
I think the type check is not smart enough, at least the `issubclass` check produces the error but it should be fine... if you return exactly `SQLModelEntity` it works...